test/evaluate.js

```diff
@@ -616,12 +633,55 @@ describe('fraud detection', () => {
       }
     ]
 
-    await runFraudDetection(1n, measurements, sparkRoundDetails)
+    await runFraudDetection(1n, measurements, sparkRoundDetails, logger)
     assert.deepStrictEqual(
       measurements.map(m => m.fraudAssessment),
       ['OK', 'IPNI_NOT_QUERIED']
     )
   })
+
+  it('rejects tasks not allowed for the node by the tasking algorithm', async () => {
```

Comment:
```suggestion
  it('rejects tasks not allowed by the tasking algorithm', async () => {
```

It's really not allowed for the node _in that round_. Don't know if we need to be so specific. I think this is clearer and simpler